### ISSUE TITLE
Cross-compilation via --{experimental_}platforms broken as of bazel 0.9.

### ISSUE THREAD

**Issue by BranLwyd** [bazelbuild/rules_go#1161]
I have a private project that I commonly cross-compile from `linux_amd64` to `linux_arm`. Yesterday, I was using bazel 0.8 with rules_go 0.8.1 and cross-compilation worked fine with a command like: `bazel build --experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_arm //:my_binary_here`. Today, I upgraded to bazel 0.9, and this command started failing with an error like the following:

```
$ bazel build --experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_arm //:my_binary_here
ERROR: /home/username/.cache/bazel/_bazel_username/d779713cc32e3b24f0e75e071a0a0628/external/io_bazel_rules_go/go/toolchain/BUILD.bazel:7:1: in platform rule @io_bazel_rules_go//go/toolchain:linux_arm: cycle in dependency graph:
    //:my_binary_here
.-> @io_bazel_rules_go//go/toolchain:linux_arm
|   @io_bazel_rules_go//go/toolchain:linux
`-- @io_bazel_rules_go//go/toolchain:linux_arm
This cycle occurred because of a configuration option
ERROR: Analysis of target '//:my_binary_here' failed; build aborted
INFO: Elapsed time: 0.196s
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Compilation works just fine if I do not pass the `--experimental_platforms` flag, though it's targeting the wrong architecture. Setting `goarch = "arm"` in the `go_binary` target does work to build the binary with the appropriate target architecture, but this is a somewhat awkward solution.

(I also tried `--platforms` instead of `--experimental_platforms` as the bazel 0.9 release notes indicate that the feature is no longer experimental, but I got the same error.)

**Comment by jayconrod** [comment]
Reproduced locally. This seems like a bug in Bazel itself. @katre, do you know what's going wrong here?

These rules are all declared in a macro, but they expand to something like this:

```
platform(
    name = "linux_arm",
    constraint_values = [
        ":linux",
        ":arm",
    ],
)

alias(
    name = "linux",
    actual = "@bazel_tools//platforms:linux",
)

alias(
    name = "arm",
    actual = "@bazel_tools//platforms:arm",
)
```

**Comment by katre** [comment]
Investigating this now

**Comment by katre** [comment]
I can also reproduce this, I'll look into debugging further.

**Comment by katre** [comment]
The problem is that [AliasRule](https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/Alias.java;l=56) itself, like all other rules, has a dependency on the current platform, so it tries to load it even though the alias is itself part of the platform.

I'll look into the best way to fix this.

**Comment by katre** [comment]
I think this should be fixed by https://bazel-review.googlesource.com/#/c/bazel/+/29630

Can someone please patch this into their Bazel and test?

**Comment by ixdy** [comment]
@katre yes, this seems to fix the issue.

**Comment by katre** [comment]
Thanks for the confirmation! I'm waiting for tests to run and we'll get this submitted.

Given the holiday, I'm not sure we can get an update out for 0.9.0, will waiting for release 0.10.0 in January be acceptable?

**Comment by jayconrod** [comment]
Also confirmed, this patch works. Thanks for the quick fix!

A point release on 0.9 would be great, but I understand if it's not possible given the holidays. rules_go doesn't depend on any features in 0.9, so we can recommend people stay on 0.8.1 until 0.10 is out. Given the monthly schedule, I'm guessing that would be some time in the Jan 15-20 range?

**Comment by katre** [comment]
According to https://github.com/bazelbuild/bazel/issues/3958 the target is first week of January.

**Comment by jayconrod** [comment]
That sounds good. Presumably the release won't be too long after the cut.

**Comment by BranLwyd** [comment]
Thank you for the quick fix! I certainly can wait until the 0.10 release as there's a workaround for the time being.
